test/CAS/Tests/AuthenticationTest.php

```diff
@@ -127,19 +127,17 @@ protected function tearDown()
      * Test that the user is redirected to the CAS server
      *
      * @return void
+     *
+     * @expectedException CAS_GracefullTerminationException
```

Comment:
I would prefer `$this->expectException()` (https://thephp.cc/news/2016/02/questioning-phpunit-best-practices and https://stackoverflow.com/questions/14561908/phpunit-doesnt-continue-test-after-expecting-an-exception/45893639#45893639).

Reply:
Me too. However, this was not introduced until PHPUnit 5.2 which is only supports PHP >= 5.6.

Here is the changelog: https://github.com/sebastianbergmann/phpunit/blob/6.4.4/ChangeLog-5.2.md#520---2016-02-05

phpCAS currently supports as far back as 5.4. Is it okay to drop these older end-of-life PHPs?

Here is the list of supported PHP versions: https://www.php.net/supported-versions.php

Reply:
We may need to discuss dropping legacy PHP support in a separate issue, and I am not sure when that would happen.

I think it's OK to raise PHPUnit requirement (since it's only for dev) but I am not sure about PHPUnit's "support" v.s. "target" towards PHP versions - I may need to look into that sometime later.

I was just intended to make sure we use latest best practices without breaking things. In this case I would say it's OK to use comment hinting.

Reply:
> I think it's OK to raise PHPUnit requirement ...

Unfortunately, this can't be done without dropping testing for PHP 5.4/5.5 as PHPUnit specifies `"php": ">=5.6"` in its `composer.json` file. So as long as the project keeps testing 5.4, we'll need to use the comment annotation.

Reply:
Ah yes, I forgot that `composer` is used to resolve PHPUnit version. Thanks!